### DISCUSSION TITLE
Increase phpcs/CI line length limit of psalm

### DISCUSSION THREAD

**Discussion by kkmuffme** [vimeo/psalm#8014] [General]
Atm CircleCI complains if line length is over 120 for psalm PRs.
This is excessively short and often leads to code that is actually less readable.

Could we increase it to 150?

**Comment by weirdan** [comment]
120 is the recommended line length per PSR-12. Also locally I have:
```console
$ echo $COLUMNS
240
```
With 120 char lines I can (almost) fit two windows alongside one another.

**Reply by kkmuffme** [reply]
https://www.php-fig.org/psr/psr-12/#:~:text=There%20MUST%20NOT%20be%20a,at%20the%20end%20of%20lines.

>Lines SHOULD NOT be longer than 80 characters

120 is acceptable if it is absolutely necessary, but 80 is preferred. If we check psalm's codebase, I'm sure at least 30% of lines are more than 80 characters.

The point of those line length limits were that screens were limited in width. That means that 1 window only fit 80/100 cols. That point is quite irrelevant today.

Even with 150 char lines I can fit two windows alongside one another too :)

**Reply by AndrolGenhald** [reply]
Fair point, but for me it doesn't make much difference. Normally I'm using an IDE with some horizontal space taken up by the file tree, if I want to look at another file to reference it I'll just stick it on another monitor. One of the places it actually matters to me is for diffing, but normally if I'm diffing it's with vim by using `git mergetool`, which has a 3 way diff at the top, so even 120 is already too much to be able to fit everything, so making it longer doesn't actually bother me much there. It could be mildly annoying for a 2 way diff, but I don't actually find myself doing that much.

**Reply by weirdan** [reply]
> The point of those line length limits were that screens were limited in width.

They are still limited on laptops. And having to move abroad when the war started made me really appreciate my 14" 1kg thinkpad.
